sdk/iot/pnp/tests/cmocka/test_az_iot_pnp_client_telemetry.c

```diff
@@ -0,0 +1,367 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "test_az_iot_pnp_client.h"
+#include <az_iot_pnp_client.h>
+#include <az_test_span.h>
+
+#include <az_precondition.h>
+#include <az_precondition_internal.h>
+
+#include <setjmp.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include <az_test_precondition.h>
+#include <cmocka.h>
+
+#define TEST_SPAN_BUFFER_SIZE 128
+
+#define TEST_DEVICE_ID_STR "my_device"
+#define TEST_DEVICE_HOSTNAME_STR "myiothub.azure-devices.net"
+#define TEST_ROOT_INTERFACE_NAME "my_root_interface_name"
+#define TEST_CONTENT_TYPE "my_content_type"
+#define TEST_CONTENT_ENCODING "my_content_encoding"
+#define TEST_COMPONENT_NAME "my_component_name"
+
+static const az_span test_device_id = AZ_SPAN_LITERAL_FROM_STR("my_device");
+static const az_span test_device_hostname = AZ_SPAN_LITERAL_FROM_STR("myiothub.azure-devices.net");
```

Comment:
Nit: any reason why you didn't use the macro's here?

Reply:
There is indeed a reason.  Which is I didn't know about them :).  Could you point me to macros you had in mind?

Reply:
Oh I meant the macro's for the strings. All the other `test` spans are using the macro's from above. Like `TEST_ROOT_INTERFACE`, `TEST_CONTENT_TYPE`, etc. So in these two cases it would be to use `TEST_DEVICE_ID_STR` and `TEST_DEVICE_HOSTNAME_STR`

Reply:
Yeah, why wasn't I consistent there?  Fixing.